cpp/src/reshape/byte_cast.cu

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright (c) 2020, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <cudf/strings/detail/utilities.hpp>
+
+#include "cudf/detail/nvtx/ranges.hpp"
+#include "cudf/replace.hpp"
+#include "cudf/strings/detail/utilities.cuh"
+
+namespace cudf {
+
+std::unique_ptr<column> byte_cast(column_view const& input_column,
+                                  rmm::mr::device_memory_resource* mr,
+                                  cudaStream_t stream)
+{
+  CUDF_FUNC_RANGE();
+  CUDF_EXPECTS(
+    cudf::is_numeric(input_column.type()) || input_column.type().id() == cudf::type_id::STRING,
+    "numeric type required to convert to big endian byte list");
+  if (input_column.type().id() == cudf::type_id::STRING) {
```

Comment:
This logic should be in a type dispatched functor as opposed to manual if checks.